x/crosschain/keeper/cctx_orchestrator_validate_outbound.go

```diff
@@ -352,21 +354,55 @@ func (k Keeper) processFailedOutboundV2(ctx sdk.Context, cctx *types.CrossChainT
 		// update status
 		cctx.SetPendingRevert("", "outbound failed")
 
+		// use a temporary context to not commit any state change in case of error
+		tmpCtx, commit := ctx.CacheContext()
+
 		// process the revert on ZEVM
-		if err := k.fungibleKeeper.ProcessV2RevertDeposit(
-			ctx,
+		to := ethcommon.HexToAddress(cctx.GetCurrentOutboundParam().Receiver)
+		evmTxResponse, err := k.fungibleKeeper.ProcessV2RevertDeposit(
+			tmpCtx,
 			cctx.InboundParams.Sender,
 			cctx.GetCurrentOutboundParam().Amount.BigInt(),
 			chainID,
 			cctx.InboundParams.CoinType,
 			cctx.InboundParams.Asset,
-			ethcommon.HexToAddress(cctx.GetCurrentOutboundParam().Receiver),
+			to,
 			cctx.RevertOptions.CallOnRevert,
 			cctx.RevertOptions.RevertMessage,
-		); err != nil {
-			return errors.Wrap(err, "failed ProcessV2RevertDeposit")
+		)
+		if fungibletypes.IsContractReverted(evmTxResponse, err) {
+			// this is a contract revert, we commit the state to save the emitted logs related to revert
+			commit()
+			return errors.Wrap(err, "revert transaction reverted")
+		} else if err != nil {
+			// this should not happen and we don't commit the state to avoid inconsistent state
+			return errors.Wrap(err, "revert transaction couldn't be processed")
 		}
 
+		// a withdrawal event in the logs could generate cctxs for outbound transactions.
+		if !evmTxResponse.Failed() {
+			logs := evmtypes.LogsToEthereum(evmTxResponse.Logs)
+			if len(logs) > 0 {
+				tmpCtx = tmpCtx.WithValue(InCCTXIndexKey, cctx.Index)
+				txOrigin := cctx.InboundParams.TxOrigin
+				if txOrigin == "" {
+					txOrigin = cctx.InboundParams.Sender
+				}
```

Comment:
Just for my understanding, how is this value going to be used? 
This would be set to the original zEVM address that created the initial withdraw ?

Reply:
This set the TX origin for the CCTX but this value should not be used for the outbound, the sender in the context is the sender field of the event